File: happyShopping_manager_web/src/main/webapp/js/service/typeTemplateService.js
//服务层
app.service('typeTemplateService',function($http){
	    	
	//读取列表数据绑定到表单中
	this.findAll=function(){
		return $http.get('../typeTemplate/findAll.action');
	};
	//分页 
	this.findPage=function(page,rows){
		return $http.get('../typeTemplate/findPage.action?page='+page+'&rows='+rows);
	};
	//查询实体
	this.findOne=function(id){
		return $http.get('../typeTemplate/findOne.action?id='+id);
	};
	//增加 
	this.add=function(entity){
		return  $http.post('../typeTemplate/add.action',entity );
	};
	//修改 
	this.update=function(entity){
		return  $http.post('../typeTemplate/update.action',entity );
	};
	//删除
	this.dele=function(ids){
		return $http.get('../typeTemplate/delete.action?ids='+ids);
	};
	//搜索
	this.search=function(page,rows,searchEntity){
		return $http.post('../typeTemplate/search.action?page='+page+"&rows="+rows, searchEntity);
	}    	
});

File: happyShopping_sellergoods_interface/src/main/java/com/happyShopping/sellergoods/service/TypeTemplateService.java
package com.happyShopping.sellergoods.service;
import java.util.List;

import com.happyShopping.model.TypeTemplate;
import entity.PageResult;
/**
 * 服务层接口
 * @author Administrator
 *
 */
public interface TypeTemplateService {

	/**
	 * 返回全部列表
	 * @return
	 */
	public List<TypeTemplate> findAll();
	
	
	/**
	 * 返回分页列表
	 * @return
	 */
	public PageResult findPage(int pageNum, int pageSize);
	
	
	/**
	 * 增加
	*/
	public void add(TypeTemplate typeTemplate);
	
	
	/**
	 * 修改
	 */
	public void update(TypeTemplate typeTemplate);
	

	/**
	 * 根据ID获取实体
	 * @param id
	 * @return
	 */
	public TypeTemplate findOne(Long id);
	
	
	/**
	 * 批量删除
	 * @param ids
	 */
	public void delete(Long[] ids);

	/**
	 * 分页
	 * @param pageNum 当前页 码
	 * @param pageSize 每页记录数
	 * @return
	 */
	public PageResult findPage(TypeTemplate typeTemplate, int pageNum, int pageSize);
	
}
